common/namespace/nsreplication/replication_task_executor.go

```diff
@@ -295,6 +296,9 @@ func (h *taskExecutorImpl) handleNamespaceUpdateReplicationTask(
 		request.Namespace.FailoverVersion = task.GetFailoverVersion()
 		request.Namespace.FailoverNotificationVersion = notificationVersion
 		request.Namespace.ReplicationConfig.FailoverHistory = ConvertFailoverHistoryToPersistenceProto(task.GetFailoverHistory())
+	} else if len(resp.Namespace.ReplicationConfig.FailoverHistory) == 0 && len(task.GetFailoverHistory()) > 0 {
+		recordUpdated = true
+		request.Namespace.ReplicationConfig.FailoverHistory = ConvertFailoverHistoryToPersistenceProto(task.GetFailoverHistory())
```

Comment:
I think the change in namespace creation path handles this issue. This logic could be hard to explain as we overwrite the failover history with a lower version. Could we change this to an error log message to see if this ever happen ?

Reply:
As discussed on Slack I've removed the out of band update, so just passing failover history on creation now.